src/box/iproto.cc

```diff
@@ -3298,6 +3298,41 @@ iproto_set_msg_max(int new_iproto_msg_max)
 	}
 }
 
+int
+iproto_session_send(struct session *session,
+		    const char *header, const char *header_end,
+		    const char *body, const char *body_end)
+{
+	struct iproto_connection *con =
```

Comment:
Assertion that checks that `session->type == SESSION_TYPE_BINARY` would be nice.

Reply:
But this function is basically the implementation of `box_iproto_send` — all contract validation is performed there.

Reply:
Yep, that is why I ask you to set an assertion here instead of check - this function can be used for internal purposes later, and good old `assert` never hurts.

Reply:
> this function can be used for internal purposes later, and good old assert never hurts.

Okay, added an assertion.